### ISSUE TITLE
Upgrading to v8.8.0 causes: Exception during Transaction#finish - ArgumentError: comparison of Float with String failed

### ISSUE THREAD

**Issue by oboxodo** [newrelic/newrelic-ruby-agent#1294]
## Description

We upgraded our agent from 6.15.0 to 8.8.0, as part of some other upgrades including Rails 6.1.x to 7.0.3. We've been testing this for over a week on our staging environment on Heroku and it's been working fine (after correcting some config issues you can see in #1276).

Now we deployed the change to our production environment and suddenly we stopped getting data in our NR transactions graph, of course affecting our Apdex and we lost visibility to our platform's performance.

We found many instances of this in our logs:

```
2022-07-29T13:04:11.941883+00:00 app[web.1]: ** [NewRelic][2022-07-29 13:04:11 +0000 web.1 (81)] ERROR : Exception during Transaction#finish
2022-07-29T13:04:11.941903+00:00 app[web.1]: ** [NewRelic][2022-07-29 13:04:11 +0000 web.1 (81)] ERROR : ArgumentError: comparison of Float with String failed
```
Something odd seems to be happening in [Transaction#finish](https://github.com/newrelic/newrelic-ruby-agent/blob/e302edc90f999288d2d710f8577111109d1e7c35/lib/new_relic/agent/transaction.rb#L513-L536) somewhere. But it didn't happen in our staging environment, which is quite similar to our production one.

Googling shows [this result](https://discuss.newrelic.com/t/argumenterror-exception-during-transaction-stop/36007/2) for the same error message, but on a different method, plus that report is for a super old version so I don't want to risk applying the suggested monkey patch.

## Expected Behavior

Transactions are properly reported and visible from NR's web UI.

## Your Environment

* ruby 3.0.4p208 (2022-04-12 revision 3fa771dded) [x86_64-linux]
* rails 7.0.3
* newrelic_rpm 8.8.0
* hosted on Heroku

## Additional context

* Similar but old and not exactly the same: https://discuss.newrelic.com/t/argumenterror-exception-during-transaction-stop/36007/2

**Comment by oboxodo** [comment]
By the way, this is our config file:

```yaml
common: &default_settings
  license_key: '<%= ENV["NEW_RELIC_LICENSE_KEY"] %>'
  app_name: <%= ENV["NEW_RELIC_APP_NAME"] %>
  log_level: info
  transaction_tracer.transaction_threshold: apdex_f

production:
  <<: *default_settings
```

**Comment by oboxodo** [comment]
We found that removing the `transaction_tracer.transaction_threshold: apdex_f` line from config fixes the problem. But that doesn't make much sense according to the config's comments which state we can set the `apdex_f` string, and that's even to be the default value.

**Comment by tannalynn** [comment]
I was able to reproduce the issue using your newrelic.yml snippet you provided and was able to determine that the agent was not checking for the apdex_f value when the configuration option was defined on one line, like your example. It was only expecting it to look like
```
    transaction_tracer:
      transaction_threshold: apdex_f 
```
I've created a PR to correct this issue since having it as a one liner shouldn't cause errors in the agent. Thank you for bringing this to our attention @oboxodo!
